collectors/proc.plugin/proc_mdstat.c

```diff
@@ -197,7 +197,7 @@ int do_proc_mdstat(int update_every, usec_t dt) {
                 }
                 s++;
             }
-            if(unlikely(str_total[0] == '\0' || str_inuse[0] == '\0')) {
+            if(unlikely(str_total[0] == '\0' || (str_inuse && str_inuse[0] == '\0'))) {

```

Comment:
This should be:
```
if (unlikely(str_total[0] == '\0' || !str_inuse || str_inuse[0] == '\0')) {
```
cause the way it is right now if `str_inuse` is `NULL` then it will succeed and do:
```
raid->inuse_disks = str2ull(str_inuse);
```

Reply:
I changed the comparison as it was suggested.